lib/cisco_node_utils/syslog_server.rb

```diff
@@ -42,12 +52,20 @@ def initialize(name,
 
     def self.syslogservers
       hash = {}
-
       syslogservers_list = config_get('syslog_server', 'server')
       return hash if syslogservers_list.nil?
 
       syslogservers_list.each do |id|
+        # The YAML regex isn't specific enough for some platforms,
+        # so we have to do further checking.
+        begin
+          IPAddr.new(id)
+        rescue
+          next
+        end
+
         level = config_get('syslog_server', 'level', id)
+        level = LEVEL_TO_NUM[level] if platform == :ios_xr
 
         vrf = config_get('syslog_server', 'vrf', id)
```

Comment:
Sounds about right but not sure that's correct for the default vrf (which I think is what is implied if vrf_arg.nil?) - for that case you should only return non-vrf cases. Also, on XR it looks like the 'vrf default' NVGENs so you may have to have slightly different behavior for XR vs NX.

```
logging 1.2.3.4
logging 2.3.4.5 vrf red
```

becomes (on XR)

```
logging 1.2.3.4 vrf default severity info port default
logging 2.3.4.5 vrf red severity info port default
```